landingai/common.py

```diff
@@ -66,7 +71,21 @@ class ClassificationPrediction(Prediction):
     """
 
 
-class OcrPrediction(Prediction):
+class BoundingBoxPrediction:
+    @property
+    @abstractproperty
+    def label(self) -> str:
+        """The text associated with this bounding box"""
+        ...
```

Comment:
i'm not sure i understand the reasoning of this change.
This property overlaps with `ClassificationPrediction`. If `label` means the predicted class here, what do you think of making the subclass, i.e. `ObjectDetectionPrediction`, inherits the `ClassificationPrediction`.

Reply:
@humpydonkey
1- OcrPrediction still inherits from `Prediction`, and has `text` attribute.
2- ObjectDetectionPrediction still inherits from `ClassificationPrediction`, and has `label`

The idea of this new `BoundingBoxPrediction` is serving as an adapter interface for all types of predictions that can be seen as "image square associated with some text". This way, we can always crop predictions from the image using the same attributes, regardless of the prediction type they are: they just need to implement `label` and `bounding_box` attributes the be compatible with the `crop_predictions` operation.

The only attribute that I suggest replacing in the future is actually the `ObjectDetectionPrediction.bboxes` because it is misleadingly in the plural (when it is actually just one bounding-box).

Reply:
> OcrPrediction still inherits from Prediction, and has text attribute.

would it create more confusion for the end users?

> The idea of this new BoundingBoxPrediction is serving as an adapter interface for all types of predictions

what about creating a separate function for the adapter interface?
sorry, i don't understand the motivation of this model chaining use case. what's the benefit of doing it this way?